files/common/usr/bin/unpack-image

```diff
@@ -96,22 +117,35 @@ tar -xzf "$UPGRADE_IMAGE_PATH" ||
 
 report_progress_inc 40 "Verifying format."
 
+# check if this is new style upgrade image
+if [[ -f "complete-unpack" ]]; then
```

Comment:
I'm fine with this name, but if we wanted to be more consistent with the nomenclature used on illumos, we could name this script `prepare`, which matches with `dx_prepare` on illumos that effectively does the same thing. That's largely where the terms `unpack` `upgrade`, `execute`, `verify`, etc. come from; i.e. `dx_unpack`, `dx_upgrade`, `dx_execute`, `dx_verify`, etc.

Ultimately, I'm good either way; this is fine with me.